torch/utils/_sympy/functions.py

```diff
@@ -156,8 +157,9 @@ def eval(cls, base, divisor):
                 return FloorDiv(base - term, divisor) + quotient
 
         try:
-            gcd = sympy.gcd(base, divisor)
-            if not equal_valued(gcd, 1):
+            # sympy.gcd tends to blow up on large sums, so use it on each summand instead
+            gcd, *gcds_ = [sympy.gcd(term, divisor) for term in base_args]
+            if not equal_valued(gcd, 1) and all(equal_valued(gcd, gcd_) for gcd_ in gcds_):
```

Comment:
These two are not equivalent right?

Reply:
Yeah I don't think they are equivalent in general, e.g. `gcd(x + 1, x + 1) = x + 1` but `gcd(x, x + 1)` and `gcd(1, x + 1)` are both `1`. Another example: `gcd(2x + 3x, 5)`. 

I actually don't know how far `sympy.gcd` goes, but the net effect here is that we'll simplify fewer expressions that could have possibly been simplified previously.

Reply:
It's a little annoying that if I have something like `(2 * x + 4 * y) // 8` I won't manage to simplify it anymore but I guess if the tests pass tis fine

Reply:
Indeed annoying. Hard to tell without data on what kind of simplifications are actually enabled in practice by `sympy.gcd`, but if its just integer factors being cancelled out then we could do something simple like: https://github.com/pytorch/pytorch/blob/7af38eb98bdceb8fc6f8635ed7dd664ef44e4b10/torch/fx/experimental/symbolic_shapes.py#L427